Fails silently with `set -o pipefail`

If I have in my `.zshrc`:
```
setopt pipe_fail  # return right-most command's non-zero return value
```

Then `__enhancd::cd` will fail silently, with `0` return value.

If I change the first lines of `cd.sh` to be:

```
function yelp () {
  # shellcheck disable=SC2154  # undeclared zsh variables in bash
  if [[ $BASH_VERSION ]]; then
    local file=${BASH_SOURCE[1]} func=${FUNCNAME[1]} line=${BASH_LINENO[0]}
  else  # zsh
    emulate -LR zsh  # Ensure 1-numbered arrays
    # $funcfiletrace has format:  file:line
    local file=${funcfiletrace[1]%:*} line=${funcfiletrace[1]##*:}
    local func=${funcstack[2]}
    [[ $func =~ / ]] && func=source  # $func may be filename. Use bash behaviour
  fi
  echo "${file##*/}:$func:$line $*" > /dev/tty
}

__enhancd::cd()
{
    trap yelp ERR
```

Then I see the following:

```
% ENHANCD_FILTER=fzy
% source init.sh
% cd
history.sh:__enhancd::history::list:25
cd.sh:__enhancd::cd:65
cd.sh:__enhancd::cd:91
%
```

The offending pipeline is at `history.sh:__enhancd::history::list:25`. If I change it to:
```
__enhancd::history::list()
{
    [[ -o pipefail ]] && local reset_pipefail=1
    set +o pipefail
    __enhancd::history::origin \
        | __enhancd::filter::reverse \
        | __enhancd::filter::unique \
        | __enhancd::filter::exists \
        | __enhancd::filter::fuzzy "$@" \
        | __enhancd::utils::grep -vx "$PWD"
    [[ $reset_pipefail ]] && set -o pipefail
}
```

Then the alias `cd` works as expected.

However, my debug then prints:
```
utils.sh:__enhancd::utils::grep:24
```

--- a/src/history.sh
+++ b/src/history.sh
@@ -22,7 +22,7 @@ __enhancd::history::list()
         | __enhancd::filter::unique \
         | __enhancd::filter::exists \
         | __enhancd::filter::fuzzy "$@" \
-        | __enhancd::utils::grep -vx "$PWD"
+        | __enhancd::utils::grep -vx "$PWD" || true
 }
 
 __enhancd::history::update()

